proxy.go

```diff
@@ -207,7 +212,7 @@ func main() {
 
 	muxAddr, err := unixAddr(channel)
 	if err != nil {
-		logger().Fatal("invalid mux socket address")
+		logger().Fatal("invalid mux socket address: ", err)
```

Comment:
I'd make use of `WithError()` like this:

```
logger().WithError(err).Fatal("invalid mux socket address")
```

Again, there are other opportunities in `main()` to make use of that call to make debugging easier.